os/kernel/sched/sched_cpuload.c

```diff
@@ -235,4 +248,61 @@ int clock_cpuload(int pid, int index, FAR struct cpuload_s *cpuload)
 	irqrestore(flags);
 	return ret;
 }
+
+#ifdef CONFIG_SCHED_CPULOAD_SNAPSHOT
+/****************************************************************************
+ * Function:  clock_cpuload_snapshot
+ *
+ * Description:
+ *   Return load measurement data for the select PID in specific time period.
+ *
+ * Parameters:
+ *   pid - The task ID of the thread of interest.  pid == 0 is the IDLE thread.
+ *   cpuload - The location to return the CPU load
+ *
+ * Return Value:
+ *   OK (0) on success; a negated errno value on failure.  The only reason
+ *   that this function can fail is if 'pid' no longer refers to a valid
+ *   thread.
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+int clock_cpuload_snapshot(int pid, FAR struct cpuload_s *cpuload)
+{
+	irqstate_t flags;
+	int tick_index;
+	int hash_index;
+
+	hash_index = PIDHASH(pid);
+	DEBUGASSERT(cpuload);
+
+	/* Check if the entry is valid (TCB field is not NULL) */
+	if (g_pidhash[hash_index].tcb == NULL) {
+		return -ESRCH;
+	}
+
+	cpuload->active = 0;
```

Comment:
There is a DEBUGASSERT for cpuload, but that is only enabled when debug is enabled. So you'd better use cpuload when cpuload is not NULL.

Reply:
Actually DEBUGASSERT is not meaningful also because it is called internally only and passed with local variable.